components/sup/src/event/types.rs

```diff
@@ -75,6 +75,7 @@ impl EventCore {
                         fqdn:          self.fqdn.clone(),
                         application:   self.application.clone(),
                         environment:   self.environment.clone(),
+                        site:          self.site.clone().unwrap_or_default(),
```

Comment:
I couldn't leave the field `site` as an `Option<String>` because the proto message requires you to have it as a `String`. 🤔 I wonder if this is the best way to do this optional field here.

Reply:
If I change this to be only `self.site.clone()` then I have the following compile error:
```
error[E0308]: mismatched types
  --> components/sup/src/event/types.rs:78:40
   |
78 |                         site:          self.site.clone(),
   |                                        ^^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found enum `std::option::Option`
   |
   = note: expected type `std::string::String`
              found type `std::option::Option<std::string::String>`
```

Reply:
This is a "feature" of v3 protobufs... the default value of a String is `""`, but v3 protobuf doesn't have a way to distinguish between an unset value and a default value (that it doesn't transmit over the wire).

In our internal types we can use `Option`, and then convert to `""` at the protobuf layer when necessary.

Reply:
Which is what I just did, right? Also I think I asked you on my previous PR so I think we are good. 💯